test_brief failures on 32-bit platforms with numpy2

### Description:

When we try to rebuild the skimage Debian package for numpy 2 on 32-bit architectures (i386, arm), the tests in `skimage/feature/tests/test_brief.py` fail:
```
__________________________ test_normal_mode[float32] ___________________________
dtype = 'float32'

    @pytest.mark.parametrize('dtype', ['float32', 'float64', 'uint8', 'int'])
    def test_normal_mode(dtype):
        """Verify the computed BRIEF descriptors with expected for normal mode."""
        img = data.coins().astype(dtype)
    
        keypoints = corner_peaks(
            corner_harris(img), min_distance=5, threshold_abs=0, threshold_rel=0.1
        )
    
        extractor = BRIEF(descriptor_size=8, sigma=2)
    
>       extractor.extract(img, keypoints[:8])

skimage/feature/tests/test_brief.py:31: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <skimage.feature.brief.BRIEF object at 0xe11eb710>
image = array(…, dtype=float32)
keypoints = array([…])

    def extract(self, image, keypoints):
[…]
        # Removing keypoints that are within (patch_size / 2) distance from the
        # image border
        self.mask = _mask_border_keypoints(image.shape, keypoints, patch_size // 2)
    
>       keypoints = np.array(
            keypoints[self.mask, :], dtype=np.int64, order='C', copy=False
        )
E       ValueError: Unable to avoid copy while creating an array as requested.
E       If using `np.array(obj, copy=False)` replace it with `np.asarray(obj)` to allow a copy when needed (no behavior change in NumPy 1.x).
E       For more details, see https://numpy.org/devdocs/numpy_2_0_migration_guide.html#adapting-to-changes-in-the-copy-keyword.

skimage/feature/brief.py:201: ValueError
```
This was reported as [Debian#1094689](https://bugs.debian.org/1094689). Full build log [here](https://buildd.debian.org/status/fetch.php?pkg=skimage&arch=i386&ver=0.24.0-7%2Bb1&stamp=1738148174&raw=0)

Removing the `copy=False` argument lets the tests pass, and from the [migration guide](https://numpy.org/devdocs/numpy_2_0_migration_guide.html#adapting-to-changes-in-the-copy-keyword) it seems that skimage here falls under the first, "Older code" argument - the copy argument is there since at least 12 years.

### Way to reproduce:

Run the CI test on a 32-bit platform using numpy 2.2.

### Version information:

```Shell
* Python 3.13.1 and 3.12.8
* Kernel: Linux 6.1.0-30-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.124-1 (2025-01-12) amd64 (x86_64)
* skimage 0.24.0 and 0.25.1
* numpy 2.2.2
```

--- a/TODO.txt
+++ b/TODO.txt
@@ -34,9 +34,8 @@ Other
 * Once NumPy 1.25.0 is minimal required version:
     * Remove optional test for a NaN-related deprecation warning from numpy.clip in
       skimage/exposure/tests/test_exposure.py::test_rescale_nan_warning
-* Delete FREE_THREADED_BUILD block in tools/github/setup_*_env.sh once free-threaded
-  wheels of scipy, numpy, cython and pywavelets are available on PyPi.
-  Last time checked (2025-02-02) only Cython was outstanding.
+* Delete FREE_THREADED_BUILD block in `.github/scripts/setup-*-env.sh` once free-threaded
+  wheels of Cython are available on PyPi.
 * Update `central_pixel` once `sp.sparse.csgraph` supports the array interface (scipy >= 1.15)
 * Finalize ``skimage.future.manual_segmentation`` API,
   see https://github.com/scikit-image/scikit-image/issues/2624

--- a/TODO.txt
+++ b/TODO.txt
@@ -1,3 +1,7 @@
+Please include necessary information to complete each entry. The
+release manager should be able to step through this file without
+external references.
+
 Version 0.27
 ------------
 * Remove deprecated `skimage.io.imshow`,
@@ -10,7 +14,6 @@ Version 0.27
   - Remove related tests, imports, etc...
 * Complete deprecation of `square`, `rectangle` and `cube` in `skimage.morphology`.
 
-
 Version 0.26
 ------------
 * In ``skimage/morphology/gray.py``, remove deprecated parameters ``shift_x``
@@ -22,22 +25,21 @@ Version 0.26
 
 Version 0.25
 ------------
-* Resolve temporary exclusion of Dask 2024.8.0 in optional dependencies, see
-  https://github.com/scikit-image/scikit-image/issues/7491 for details.
 
 Other
 -----
 * Remove warningsfilter for imageio and Pillow once
   https://github.com/python-pillow/Pillow/pull/7125 is shipped in the minimal required
   version of pillow (probably the release after Pillow 9.5.0).
-* NumPy 1.25.0 is minimal required version:
+* Once NumPy 1.25.0 is minimal required version:
     * Remove optional test for a NaN-related deprecation warning from numpy.clip in
       skimage/exposure/tests/test_exposure.py::test_rescale_nan_warning
+* Delete FREE_THREADED_BUILD block in tools/github/setup_*_env.sh once free-threaded
+  wheels of scipy, numpy, cython and pywavelets are available on PyPi.
+  Last time checked (2025-02-02) only Cython was outstanding.
+* Update `central_pixel` once `sp.sparse.csgraph` supports the array interface (scipy >= 1.15)
 * Finalize ``skimage.future.manual_segmentation`` API,
   see https://github.com/scikit-image/scikit-image/issues/2624
-* Delete FREE_THREADED_BUILD block in tools/github/before_install.sh once free-threaded
-  wheels for scipy, numpy, cython and pywavelets, ... are available on PyPi
-* Update `central_pixel` once `sp.sparse.csgraph` supports the array interface
 
 Post numpy 2
 ------------
@@ -52,3 +54,4 @@ Post numpy 2
   `nightly_wheel_build.yml`.
 - Remove references to `numpy.bool8` once NumPy 2.0 is minimal required version.
 - scipy>=1.12: remove SCIPY_CG_TOL_PARAM_NAME in `_shared.compat.py`.
+- Remove `np2` check in `skimage/feature/brief.py`.

--- a/skimage/feature/brief.py
+++ b/skimage/feature/brief.py
@@ -199,7 +199,10 @@ class BRIEF(DescriptorExtractor):
         self.mask = _mask_border_keypoints(image.shape, keypoints, patch_size // 2)
 
         keypoints = np.array(
-            keypoints[self.mask, :], dtype=np.int64, order='C', copy=False
+            keypoints[self.mask, :],
+            dtype=np.int64,
+            order='C',
+            copy=None,
         )
 
         self.descriptors = np.zeros(

--- a/skimage/feature/brief.py
+++ b/skimage/feature/brief.py
@@ -1,6 +1,7 @@
 import copy
 
 import numpy as np
+from packaging.version import Version
 
 from .._shared.filters import gaussian
 from .._shared.utils import check_nD
@@ -10,6 +11,9 @@ from .util import (
     _mask_border_keypoints,
     _prepare_grayscale_input_2D,
 )
+
+
+np2 = Version(np.__version__) >= Version('2')
 
 
 class BRIEF(DescriptorExtractor):
@@ -202,7 +206,7 @@ class BRIEF(DescriptorExtractor):
             keypoints[self.mask, :],
             dtype=np.int64,
             order='C',
-            copy=None,
+            copy=None if np2 else False,
         )
 
         self.descriptors = np.zeros(

